test/test_torch.py

```diff
@@ -18181,6 +18181,57 @@ def test_large_linspace(self, device, dtype):
         x = torch.linspace(start, end, steps, dtype=dtype, device=device)
         self.assertGreater(x[1] - x[0], (end - start) / steps)
 
+    @dtypes(torch.int64, torch.float, torch.complex128)
+    def test_movedim_invalid(self, device, dtype):
+        shape = self._rand_shape(4, min_size=5, max_size=10)
+        x = self._generate_input(shape, dtype, device, False)
+
+        # Invalid `src` and `dst` dimension
+        with self.assertRaisesRegex(IndexError, "Dimension out of range"):
+            torch.movedim(x, 5, 0)
+
+        with self.assertRaisesRegex(IndexError, "Dimension out of range"):
+            torch.movedim(x, 0, 5)
+
+        # Mismatch in size of `src` and `dst`
+        with self.assertRaisesRegex(RuntimeError, "movedim: Invalid source or destination dims:"):
+            torch.movedim(x, (1, 0), (0, ))
+
+        with self.assertRaisesRegex(RuntimeError, "movedim: repeated dim in `src`"):
+            torch.movedim(x, (0, 0), (0, 1))
+
+        with self.assertRaisesRegex(RuntimeError, "movedim: repeated dim in `dst`"):
+            torch.movedim(x, (0, 1), (1, 1))
+
+    @unittest.skipIf(not TEST_NUMPY, "NumPy not found")
+    @dtypes(torch.int64, torch.float, torch.complex128)
+    def test_movedim(self, device, dtype):
+        for nd in range(5):
+            shape = self._rand_shape(nd, min_size=5, max_size=10)
+            x = self._generate_input(shape, dtype, device, with_extremal=False)
+            for random_negative in [True, False]:
+                for src_dim, dst_dim in permutations(range(nd), r=2):
+                    if random_negative:
+                        src_dim = src_dim - nd
+                    # Integer Inputs
+                    torch_fn = partial(torch.movedim, src=src_dim, dst=dst_dim)
+                    np_fn = partial(np.moveaxis, source=src_dim, destination=dst_dim)
+                    self.compare_with_numpy(torch_fn, np_fn, x, device=None, dtype=None)
+
+                if nd > 0:
+                    for src_sequence in permutations(range(nd), r=random.randint(1, nd)):
+                        # Randomly change a dim to a negative dim representation of itself.
+                        if random_negative:
+                            random_idx = random.randint(0, len(src_sequence) - 1)
+                            src_sequence = list(src_sequence)
+                            src_sequence[random_idx] = src_sequence[random_idx] - nd
+                            src_sequence = tuple(src_sequence)
+                        # Sequence Inputs
+                        dst_sequence = tuple(random.sample(range(nd), len(src_sequence)))
+                        torch_fn = partial(torch.movedim, src=src_sequence, dst=dst_sequence)
+                        np_fn = partial(np.moveaxis, source=src_sequence, destination=dst_sequence)
+                        self.compare_with_numpy(torch_fn, np_fn, x, device=None, dtype=None)
+
```

Comment:
One edge case I can think of is if if "movedim" performs a no-op. For example,
```
x = torch.randn(2, 3, 5, 7, 11)
torch.movedim(x, (0, 1), (0, 1))
```
Could we add an explicit test to test this?

Reply:
Makes sense. Thanks.

Reply:
Done.